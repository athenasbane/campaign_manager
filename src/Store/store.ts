import { configureStore } from "@reduxjs/toolkit";
import modalReducer from "./slices/modals";
import listsReducer from "./slices/lists";
import sessionsReducer from "./slices/sessions";
import mapsReducer from "./slices/map";
import contentReducer from "./slices/content";

const store = configureStore({
  reducer: {
    modals: modalReducer,
    lists: listsReducer,
    sessions: sessionsReducer,
    maps: mapsReducer,
    content: contentReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

export default store;
