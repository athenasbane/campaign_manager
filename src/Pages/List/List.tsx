import MuiList from "@mui/material/List";
import Typography from "@mui/material/Typography";
import ListItem from "@mui/material/ListItem";
import { useParams } from "react-router-dom";
import { useAppSelector } from "hooks/store.hooks";
import { selectFilteredList } from "Store/slices/lists";
import Link from "Components/Atom/Link/Link";

export default function List() {
  const { slug } = useParams();
  const lists = useAppSelector((state) =>
    selectFilteredList(state.lists, slug)
  );

  const list = lists.map((list) => {
    return (
      <MuiList key={list.param}>
        <ListItem>
          {list.parentDisplayLabel ? (
            <Typography align="center" variant="h2">
              {list.parentDisplayLabel}
            </Typography>
          ) : null}
        </ListItem>
        <ListItem>
          <Typography align="center" variant="h3">
            {list.displayLabel}
          </Typography>
        </ListItem>
        {list.links.map((link) => {
          return (
            <ListItem key={link.value}>
              <Link
                typographyVariant="h5"
                linkDisplayLabel={link.displayLabel}
                route={link.path}
              />
            </ListItem>
          );
        })}
      </MuiList>
    );
  });

  return <>{list}</>;
}
