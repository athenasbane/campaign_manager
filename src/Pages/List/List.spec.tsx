import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import List from "../../Pages/List/List";
import { Provider } from "react-redux";
import store from "../../Store/store";

describe("Page - List", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <List />
      </Provider>,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });
});
