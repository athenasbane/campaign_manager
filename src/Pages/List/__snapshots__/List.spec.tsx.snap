// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - List should match snapshot 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    />
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
      >
        Maps
      </h3>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/map/tordenhelm"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Tordenhelm
        </h5>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/list/the_nals"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          The Nals
        </h5>
      </a>
    </li>
  </ul>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
      >
        Maps
      </h2>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
      >
        The Nals
      </h3>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/list/the_nals_ctv"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Cities, Towns & Villages
        </h5>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/map/cirrain_coast"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Cirrane Coast
        </h5>
      </a>
    </li>
  </ul>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
      >
        Maps
      </h2>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
      >
        Cities, Towns & Villages
      </h3>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/map/klilcaithness"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Klilcaithness
        </h5>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/map/shambles"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          The Shambles
        </h5>
      </a>
    </li>
  </ul>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
      >
        Lore
      </h2>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
      >
        Basic Overview
      </h3>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/content/alignment"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Alignment
        </h5>
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <a
        href="/content/dead_gods"
        style="text-decoration: none;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="link_typography"
        >
          Dead Gods
        </h5>
      </a>
    </li>
  </ul>
</div>
`;
